### PR TITLE
[4.2.01] Fix msvc cuda release

### PR DESCRIPTION
**Pull request by crtrott**
Equivalent to #6659

### CODE REVIEW

File: core/src/Serial/Kokkos_Serial.hpp [hunk 1]
```diff
@@ -30,6 +30,7 @@ static_assert(false,
 
 #include <cstddef>
 #include <iosfwd>
+#include <iterator>
```
Comment (fnrizzi) [inline core/src/Serial/Kokkos_Serial.hpp:33]: why is this needed here?

**Comment by crtrott** [comment]
Retest this please

File: core/src/Serial/Kokkos_Serial.hpp (hunk 1 above)
Comment (crtrott) [reply core/src/Serial/Kokkos_Serial.hpp:33]: On Windows with NVCC the implicit includes are different exposing a missing include I beleive it was for `std::back_inserter`

Comment (masterleinad) [reply core/src/Serial/Kokkos_Serial.hpp:33]: See https://github.com/kokkos/kokkos/pull/6659#discussion_r1421613190.

**Review by masterleinad — APPROVED** [review]

**Comment by crtrott** [comment]
Retest this please.

**Comment by dalg24** [comment]
Retest this please

**Comment by crtrott** [comment]
Retest this please.
